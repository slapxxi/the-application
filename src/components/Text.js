// @flow
import _ from 'lodash';
import styled from 'react-emotion';

const Text = styled('p')`
  max-width: 45em;
  text-overflow: ellipsis;
  font-size: 15px;
  line-height: 1.6;
  color: ${(p) => _.get(p, 'theme.colors.text')};
`;

export default Text;
