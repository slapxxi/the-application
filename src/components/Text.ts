import * as _ from 'lodash';
import styled from 'react-emotion';

const Text = styled('p')`
  text-overflow: ellipsis;
  font-size: 15px;
  line-height: 1.6;
  color: ${(p) => _.get(p, 'theme.colors.text')};
`;

export default Text;
