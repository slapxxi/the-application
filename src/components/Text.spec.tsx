import React from 'react';
import renderer from 'react-test-renderer';
import { render } from 'react-testing-library';
import Text from './Text';

const theme = {
  colors: {
    text: 'hotpink',
  },
};

it('renders', () => {
  const component = renderer.create(<Text>sample text</Text>);
  expect(component).toMatchSnapshot();
});

it('accepts `theme` prop', () => {
  const component = renderer.create(
    <Text theme={theme}>sample text</Text>,
  );
  expect(component).toMatchSnapshot();
});

it('does not throw if `theme` is malformed', () => {
  expect(() => render(<Text theme={{}}>text</Text>)).not.toThrow();
});
