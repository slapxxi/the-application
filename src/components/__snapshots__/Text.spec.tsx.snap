// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`accepts \`theme\` prop 1`] = `
.emotion-0 {
  text-overflow: ellipsis;
  font-size: 15px;
  line-height: 1.6;
  color: black;
}

<p
  class="emotion-0"
>
  text
</p>
`;

exports[`renders 1`] = `
.emotion-0 {
  text-overflow: ellipsis;
  font-size: 15px;
  line-height: 1.6;
}

<p
  class="emotion-0"
>
  sample text
</p>
`;
